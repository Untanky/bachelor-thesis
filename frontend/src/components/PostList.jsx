import React from 'react';
import PropTypes from 'prop-types';

import Post from './Post';

import './PostList.scss';

const PostList = ({ posts }) => {
  return (
    <div className="post-list">
      { posts.map((post) => (
        <Post 
          key={Math.random()}
          post={post}
        />
      )) }
    </div>
  )
};

PostList.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      description: PropTypes.string,
      id: PropTypes.number,
    })
  ).isRequired,
};

export default PostList;
