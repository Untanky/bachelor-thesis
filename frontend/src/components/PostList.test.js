import React from 'react';
import PostList from './PostList';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

const postList = [
  {
    id: 1,
    title: 'Test title',
    description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
    numberOfComments: 6,
    url: 'post/1'
  },
  {
    id: 2,
    title: 'My newest post',
    description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
    numberOfComments: 0,
    url: 'post/2'
  },
  {
    id: 3,
    title: 'Post with some content',
    description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
    numberOfComments: 200,
    url: 'post/3'
  },
];

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <PostList posts={postList} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
