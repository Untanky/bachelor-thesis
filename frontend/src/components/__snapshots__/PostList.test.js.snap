// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="post-list"
>
  <div
    className="post"
  >
    <h2
      className="post-title"
    >
      Test title
    </h2>
    <p
      className="post-description"
    >
      This is the description for the test post. It contains vital data for the post and puts it into context.
    </p>
    <div
      className="button-container"
    >
      <a
        className="button edit-link"
        href="/post/edit/1"
        onClick={[Function]}
      >
        Edit Post
      </a>
      <button
        className="button delete delete-link"
      >
        Delete post
      </button>
    </div>
  </div>
  <div
    className="post"
  >
    <h2
      className="post-title"
    >
      My newest post
    </h2>
    <p
      className="post-description"
    >
      This is the description for the test post. It contains vital data for the post and puts it into context.
    </p>
    <div
      className="button-container"
    >
      <a
        className="button edit-link"
        href="/post/edit/2"
        onClick={[Function]}
      >
        Edit Post
      </a>
      <button
        className="button delete delete-link"
      >
        Delete post
      </button>
    </div>
  </div>
  <div
    className="post"
  >
    <h2
      className="post-title"
    >
      Post with some content
    </h2>
    <p
      className="post-description"
    >
      This is the description for the test post. It contains vital data for the post and puts it into context.
    </p>
    <div
      className="button-container"
    >
      <a
        className="button edit-link"
        href="/post/edit/3"
        onClick={[Function]}
      >
        Edit Post
      </a>
      <button
        className="button delete delete-link"
      >
        Delete post
      </button>
    </div>
  </div>
</div>
`;
